### PR TITLE
Fix lr schedule in fine-tuning + add documentation

### PR DESCRIPTION
**Pull request by nostalgebraist**
Fixes an issue where `device_train.py` didn't use the lr schedule params in the config correctly.

There are several notions of global step in this repo:

1. numbers in checkpoint paths
2. step in `meta.json`
3. counts stored in optax transforms (one for each transform with step state)

**Currently:**

`device_train.py` resets 1 and 2 to zero, but kept the step values in 3.

The schedule continues from the loaded step, which generally means finetuning happens at `end_lr`.

**This PR:**

I reset the optax step value for the lr schedule to zero.  This will give the expected warmup/anneal behavior from the config.

There's also a global step in the `scale_by_adam` state, for bias correction.  I don't reset this, as we're loading the accumulators and they don't need bias correction

----
Also added/modified some documentation strings to address common questions from Discord.

### CODE REVIEW

**Comment by kingoflolz** [comment]
Thanks!
